### PR TITLE
[docs] Documentation for CIRCT round-trip.

### PR DESCRIPTION
**Pull request by cgyurgyik**

### CODE REVIEW

File: docs/fud/circt.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# CIRCT
+
+An ongoing effort is under way to establish Calyx as a Dialect in the LLVM umbrella project [CIRCT][].
```
Comment (sampsyo) [inline docs/fud/circt.md:3]:
```suggestion
An ongoing effort is under way to establish Calyx as a *dialect* in the LLVM umbrella project [CIRCT][]. 
```

I'd probably not capitalize "Dialect" throughout—I don't think people tend to use it as a proper noun.

File: docs/fud/circt.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# CIRCT
+
+An ongoing effort is under way to establish Calyx as a Dialect in the LLVM umbrella project [CIRCT][]. 
+The Calyx Dialect documentation is found [here][calyx-dialect]. While semantically equivalent, they
```
Comment (sampsyo) [inline docs/fud/circt.md:4]:
```suggestion
There is documentation about the Calyx dialect [on the MLIR site][calyx-dialect]. While semantically equivalent, they
```

File: docs/fud/circt.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+all the optimizations that the native Rust compiler supports, we have crafted an emitter from the Calyx 
+Dialect (MLIR) to the native compiler representation (used by the Rust compiler). This means you can 
+lower from your favorite frontend in MLIR to the Calyx Dialect, and continue all the way to SystemVerilog
+(with spunky optimizations) using the native compiler.
+
+The native compiler also supports round-tripping back into the MLIR representation. We'll assume you've 
+already built the Rust compiler and installed `fud`. Here are the steps below to round-trip:
+
+## MLIR -> Rust compiler
+1. Set up the CIRCT project; instructions found [here][circt-setup].
```
Comment (sampsyo) [inline docs/fud/circt.md:15]:
```suggestion
1. Set up the CIRCT project with [these instructions][circt-setup].
```

**Review by sampsyo — COMMENTED** [review]
Excellent! Super useful to have this basic information in the docs. :+1: Just a couple of suggestions within.

**Comment by rachitnigam** [comment]
Merge when @sampsyo's comments are addressed.

**Comment by cgyurgyik** [comment]
@sampsy

> Excellent! Super useful to have this basic information in the docs. 👍 Just a couple of suggestions within.

Good suggestions thanks!

**Pull request merged**